amazon-kinesis-connector-flink/src/main/java/software/amazon/kinesis/connectors/flink/internals/KinesisDataFetcher.java

```diff
@@ -1222,14 +1224,19 @@ public void onProcessingTime(long timestamp) {
 	 *
 	 * @return a {@link MetricGroup} that can be used to update metric values
 	 */
-	private MetricGroup registerShardMetricGroup(final MetricGroup metricGroup, final KinesisStreamShardState shardState) {
-		return metricGroup
-				.addGroup(
-						KinesisConsumerMetricConstants.STREAM_METRICS_GROUP,
-						shardState.getStreamShardHandle().getStreamName())
-				.addGroup(
-						KinesisConsumerMetricConstants.SHARD_METRICS_GROUP,
-						shardState.getStreamShardHandle().getShard().getShardId());
+	public static GenericMetricGroup registerShardMetricGroup(final MetricGroup metricGroup, final StreamShardHandle shardHandle) {
+		try {
+			return (GenericMetricGroup) metricGroup
+					.addGroup(
+							KinesisConsumerMetricConstants.STREAM_METRICS_GROUP,
+							shardHandle.getStreamName())
+					.addGroup(
+							KinesisConsumerMetricConstants.SHARD_METRICS_GROUP,
+							shardHandle.getShard().getShardId());
+		} catch (ClassCastException e) {
+			throw new ClassCastException("MetricGroup provided is not a GenericMetricGroup. Please ensure the" +
```

Comment:
I am worried that if the implementation of `MetricGroup` changes then the connector will not work. To make this more future proof, it would be best to perform the check in the `unregister` method later. Sorry this contradicts what I said earlier. 

```
public void unregister() {
    if (metricGroup instanceof AbstractMetricsGroup) {
        ((AbstractMetricsGroup) this.metricGroup.close());
    }
}
```

Reply:
`AbstractMetricsGroup` is indeed more future proof, thanks!